devicetypes/smartthings/zigbee-switch.src/zigbee-switch.groovy

```diff
@@ -19,6 +19,7 @@ metadata {
 		capability "Refresh"
 		capability "Switch"
 		capability "Health Check"
+		capability "Battery"
 
 		// Generic
 		fingerprint profileId: "C05E", deviceId: "0000", inClusters: "0006", deviceJoinName: "Light", ocfDeviceType: "oic.d.light" //Generic On/Off Light
```

Comment:
your switch needs to specify a unique metadata that has support for displaying the battery, or none of these changes will display in the app

Reply:
so, what you need is to specify a the `mnmn` and `vid` fields for your specific device's fingerprint. Here's an example that has done that: https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/27169c8332f7acef18b74e0af2fdf9693e83df5e/devicetypes/smartthings/ozom-smart-siren.src/ozom-smart-siren.groovy#L30

but you would have to upload your own ui-metadata using the API.